AsyncDisplayKit/ASCellNode.mm

```diff
@@ -117,15 +118,18 @@ - (void)layoutDidFinish
   _viewControllerNode.frame = self.bounds;
 }
 
-- (void)__setNeedsLayout
+- (void)displayNodeDidInvalidateSizeOldSize:(CGSize)oldSize
 {
-  CGSize oldSize = self.calculatedSize;
-  [super __setNeedsLayout];
-  
-  //Adding this lock because lock used to be held when this method was called. Not sure if it's necessary for
-  //didRelayoutFromOldSize:toNewSize:
-  ASDN::MutexLocker l(__instanceLock__);
-  [self didRelayoutFromOldSize:oldSize toNewSize:self.calculatedSize];
+  ASSizeRange constrainedSize = ASSizeRangeMake(CGSizeZero, CGSizeMake(CGRectGetWidth(self.bounds), CGFLOAT_MAX));
+  if (_interactionDelegate != nil) {
+    constrainedSize = [_interactionDelegate constrainedSizeForNode:self];
+  }
+  CGSize newSize = [self layoutThatFits:constrainedSize].size;
+    
+  if (CGSizeEqualToSize(oldSize, newSize) == NO) {
+    self.frame = {self.frame.origin, newSize};
```

Comment:
Can we actually remove this self.frame line?  I THINK once we inform UIKit of the change, then the _ASCollectionViewCell etc will be updated - which has a layoutSubviews that sets _node.frame = self.bounds.

If so, we should remove this line.  If removing it causes us to hit extra layout passes due to bounds != layout.size, then we'll have to keep it.  But I think layoutSubviews should be called top-down and so the frame would be set on the cell before going into -layout to apply the ASLayout to subnodes.